### PR TITLE
Accessibility: Add support for `accessibility-action-*` callback

### PR DESCRIPTION
**Pull request by ogoffart**
Merge of the feature/accessibility branch

contains 
- #4824
- #4959
- #5062
- #5065
- #5066

### CODE REVIEW

File: internal/backends/qt/qt_accessible.rs [hunk 1]
```diff
@@ -616,6 +622,8 @@ cpp! {{
         void *interface_cast(QAccessible::InterfaceType t) override {
             if (t == QAccessible::ValueInterface && !item_string_property(m_data, QAccessible::Value).isEmpty()) {
                 return static_cast<QAccessibleValueInterface*>(this);
+            } else if (t == QAccessible::ActionInterface && !item_string_property(m_data, QAccessible::Value).isEmpty()) {
```
Comment (tronical) [inline internal/backends/qt/qt_accessible.rs:625]: Hm, I'm wondering if the value check is correct.

Actions like increment or decrement certainly require a value, but there are also actions that make sense on accessible widgets that clearly don't have a value, such as scrolling actions on a scrollview, or showing/hiding of a tooltip.

```suggestion
            } else if (t == QAccessible::ActionInterface) {
```

**Review by tronical — APPROVED** [review]

File: internal/backends/qt/qt_accessible.rs (hunk 1 above)
Comment (ogoffart) [reply internal/backends/qt/qt_accessible.rs:625]: indeed.  bad copy paste.

File: internal/compiler/typeregister.rs [hunk 2]
```diff
@@ -98,18 +98,32 @@ pub const RESERVED_ROTATION_PROPERTIES: &[(&str, Type)] = &[
...
+        ("accessible-value", Type::String),
+        ("accessible-value-maximum", Type::Float32),
+        ("accessible-value-minimum", Type::Float32),
+        ("accessible-value-step", Type::Float32),
+        ("accessible-action-default", Type::Callback { return_type: None, args: vec![] }),
+        ("accessible-action-increment", Type::Callback { return_type: None, args: vec![] }),
+        ("accessible-action-decrement", Type::Callback { return_type: None, args: vec![] }),
+        (
+            "accessible-action-set-value",
+            Type::Callback { return_type: None, args: vec![Type::Float32] },
```
Comment (ogoffart) [inline internal/compiler/typeregister.rs:118]: I wonder if the action should take a float or a string here.
It seems that the value on a line edit would be a string.
also the value property is a string.

**Comment by ogoffart** [comment]
Merged manually in 13127dc71cc850813384cf50445fd8da798e0764

**Pull request closed without merging**